tests/e2e/utils.js

```diff
@@ -52,6 +44,23 @@ export function getExtensionElement(id) {
   return $(`>>>[data-qa="${id}"]`);
 }
 
+export async function waitForIdleBackgroundTasks() {
+  const protocol = browser.isChromium ? 'chrome-extension' : 'moz-extension';
+
+  if (!(await browser.getUrl()).startsWith(protocol)) {
```

Comment:
maybe instead of throwing we ensure that this is executes in the right context. How about we open new browser window with extension page and close it after we get the response?

(trick with the window I think is needed only for Firefox)

Reply:
This code might be called before onboarding, in the panel contexts, and how it can open url/window is related to that context - to simply not break tests (as jumping on windows/url makes test flaky). This is the reason why I did not add that into the function, but there is a check to be sure, if someone would use it on wrong context.